orchagent/pfcwdorch.h

```diff
@@ -95,7 +97,13 @@ class PfcWdSwOrch: public PfcWdOrch<DropHandler, ForwardHandler>
 
     string filterPfcCounters(string counters, set<uint8_t>& losslessTc);
     string getFlexCounterTableKey(string s);
+
+    void disableBigRedSwitchMode();
+    void enableBigRedSwitchMode();
+    void setBigRedSwitchMode(string value);
+
     map<sai_object_id_t, PfcWdQueueEntry> m_entryMap;
+    map<sai_object_id_t, PfcWdQueueEntry> m_brsEntryMap;
```

Comment:
Why separate map for BRS?

Reply:
After disable big-red-switch, we need to keep the PFC WD original configuration. This information actually is stored in m_entryMap. It would be better if we don't mix this information with brs.

Reply:
In line 398 in .cpp file entry is removed from the regular map when BRS is enabled, but it is not added back when BRS is disabled.
In case we had some handler, and BRS is flipped on/off, will we end up with no handler after BRS off?

Reply:
That’s true. The previous active entries mark the storm happens when BRS was enabled. After BRS runs for a while, the previous state is out of date so it’s unecessary to create those handlers. Instead we just need to recover the PWCWD configuration and let it detect storm. we need is just the configuration which is the entrymap

Reply:
ok